core/lib/dal/migrations/20240112194527_snapshots_applier_processed_chunks_format_change.up.sql

```diff
@@ -0,0 +1,9 @@
+ALTER TABLE snapshot_recovery DROP COLUMN last_finished_chunk_id;
+ALTER TABLE snapshot_recovery DROP COLUMN total_chunk_count;
+
+ALTER TABLE snapshot_recovery ADD COLUMN storage_logs_chunks_ids_to_process INT[] NOT NULL;
+ALTER TABLE snapshot_recovery ADD COLUMN storage_logs_chunks_ids_already_processed INT[] NOT NULL;
```

Comment:
yes, I've written applier in a way that assumes that chunk_ids are unique and nothing more, just as a defensive mechanism. I agree having a boolean array has nice properties, but I think I can also minimize the risks of invalid state in dal (I will introduce a separate method that 'marks one chunk as complete') and it will have some guard mechanisms. For me those two arrays just look a bit more readable, though it's highly personal

Reply:
I've actually changed my mind and changed those two arrays into a bool array